crates/mfem/examples/ex1.rs

```diff
@@ -96,33 +80,32 @@ fn main() -> anyhow::Result<()> {
     //    the boundary attributes from the mesh as essential (Dirichlet) and
     //    converting them to a list of true dofs.
     let mut ess_tdof_list = ArrayInt::new();
-    if let Some(max_bdr_attr) = mesh.get_bdr_attributes().iter().max() {
+    if let Some(max_bdr_attr) = fespace.mesh().bdr_attributes().iter().max() {
         let mut ess_bdr = ArrayInt::with_len(*max_bdr_attr as usize);
-        ess_bdr.set_all(1);
+        ess_bdr.fill(1);
         fespace.get_essential_true_dofs(&ess_bdr, &mut ess_tdof_list, None);
     }
 
-    // 7. Set up the linear form b(.) which corresponds to the right-hand side of
-    //    the FEM linear system, which in this case is (1,phi_i) where phi_i are
-    //    the basis functions in the finite element fespace.
+    // 7. Set up the linear form b(.) which corresponds to the
+    //    right-hand side of the FEM linear system, which in this case
+    //    is (1,phi_i) where phi_i are the basis functions in the
+    //    finite element `fespace`.
+    let mut one = ConstantCoefficient::new(1.0);
     let mut b = LinearForm::new(&fespace);
-    let one = ConstantCoefficient::new(1.0);
-    let integrator = DomainLFIntegrator::new(&one, 2, 0);
-    b.add_domain_integrator(integrator);
+    b.add_domain_integrator(DomainLFIntegrator::new(&mut one));
     b.assemble();
 
     // 8. Define the solution vector x as a finite element grid function
     //    corresponding to fespace. Initialize x with initial guess of zero,
     //    which satisfies the boundary conditions.
     let mut x = GridFunction::new(&fespace);
-    x.set_all(0.0);
+    x.fill(0.0);
 
     // 9. Set up the bilinear form a(.,.) on the finite element space
-    //    corresponding to the Laplacian operator -Delta, by adding the Diffusion
-    //    domain integrator.
+    //    corresponding to the Laplacian operator -Delta, by adding
+    //    the Diffusion domain integrator.
     let mut a = BilinearForm::new(&fespace);
-    let bf_integrator = DiffusionIntegrator::new(&one);
-    a.add_domain_integrator(bf_integrator);
+    a.add_domain_integrator(DiffusionIntegrator::new());
```

Comment:
Even though the outcome is the same, let's keep the explicit coefficient here, because it's supposed to be a direct translation of `ex1.cpp` from MFEM.
http://github.com/mfem/mfem/blob/dc9128ef596e84daf1138aa3046b826bba9d259f/examples/ex1.cpp#L203

```suggestion
    a.add_domain_integrator(DiffusionIntegrator::new(&mut one));
```

Reply:
Ok but we need to duplicate the `one` for it is already mutably borrowed.  (It seems the same should be desirable in C++ because the coefficient can be mutated but the circumstances under which it is are not clear.)